x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupIndexerAction.java

```diff
@@ -186,5 +198,32 @@ protected void finishHim() {
                 listener.onFailure(e);
             }
         }
+
+        @Override
+        protected void onOperation(@Nullable ShardRouting shard, final ShardIterator shardIt, int shardIndex, Exception e) {
+            // when this shard operation failed, cancel other shard operations
+            cancelOtherShardIndexers();
```

Comment:
In case a shard has replicas the behaviour of the super class is to retry on the replica shard.
With the cancel logic, I don't think we want this?

Reply:
When do you want cancelOtherShardIndexers to be called? I change the code, when the shard failed, to avoid retry, I set the shardIt to the last shard.

Reply:
> I change the code, when the shard failed, to avoid retry, I set the shardIt to the last shard.

👍